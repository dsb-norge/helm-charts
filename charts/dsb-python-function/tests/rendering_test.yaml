suite: test deployment
tests:
  - it: Minimal manifest should match snapshot
    templates:
      - deployment.yaml
      - pod-disruption-budget.yaml
      - service.yaml
    set:
      image: python
      tag: latest
    asserts:
      - matchSnapshot: { }
  - it: should render default with PodDisruptionBudget and should match snapshot
    templates:
      - pod-disruption-budget.yaml
    asserts:
      - matchSnapshot: { }
      - isKind:
          of: PodDisruptionBudget
  - it: should be possible to render without PodDisruptionBudget
    templates:
      - pod-disruption-budget.yaml
    set:
      createPodDisruptionBudget: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should not create PodDisruptionBudget when replicas is 0
    templates:
      - pod-disruption-budget.yaml
    set:
      replicas: 0
    asserts:
      - hasDocuments:
          count: 0
  - it: Full manifest should match snapshot
    templates:
      - config-map.yaml
      - deployment.yaml
      - ingress.yaml
      - pod-disruption-budget.yaml
      - service.yaml
      - serviceMonitor.yaml
    set:
      replicas: 2
      minAvailableReplicas: 1
      image: dsb/py-fn
      tag: 1.0.0
      deploymentAnnotations:
        key1: value1
        key2: value2
      resourceName: "pyfn1"
      config:
        SOME_URL: "https://www.dsb.no"
        root:
          key1: value1
          key2: value2
      secretRefs:
        - secret1
      configMapRefs:
        - configMap1
      extraVolumes:
        - name: my-empty-dir
          emptyDir: {}
      extraVolumeMounts:
        - name: my-empty-dir
          mountPath: /work
      ingress_host: example.com
      ingress_path: /example
      port: 8000
      memory_request: "64Mi"
      memory_limit: "128Mi"
      cpu_request: "25m"
      cpu_limit: "100m"
      prometheus_enabled: "false"
      lang: "en_GB.UTF-8"
      dashboard:
        name: "Rendering test"
        component: "Function"
        partOf: "Verification stuff"
        createdBy: "https://github.com/dsb-norge/helm-charts"
    asserts:
      - matchSnapshot: { }
  - it: Support multiple ingress hosts
    templates:
      - ingress.yaml
      - service.yaml
    set:
     image: python
     tag: latest
     prometheus_enabled: false
     ingress:
        - host: example.com
        - host: admin.example.com
          path: /secret
    asserts:
      - matchSnapshot: { }
