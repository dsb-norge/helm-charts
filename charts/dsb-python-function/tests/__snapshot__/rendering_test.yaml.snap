Full manifest should match snapshot:
  1: |
    apiVersion: v1
    data:
      SOME_URL: https://www.dsb.no
      root.key1: value1
      root.key2: value2
    kind: ConfigMap
    metadata:
      name: pyfn1
      namespace: NAMESPACE
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        key1: value1
        key2: value2
      name: pyfn1
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          app: pyfn1
      template:
        metadata:
          annotations:
            apparmor.security.beta.kubernetes.io/pod: runtime/default
            checksum: chart-version=0.1.0_config-hash=71e97e317fa75ec51f11dcfcd7f01f81da4ab68e828e265599b1d4dc2bd4a65b
          labels:
            app: pyfn1
            app.kubernetes.io/component: Function
            app.kubernetes.io/created-by: https://github.com/dsb-norge/helm-charts
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: Rendering test
            app.kubernetes.io/part-of: Verification stuff
            app.kubernetes.io/version: 1.0.0
            azure.workload.identity/use: "false"
            helm.sh/chart: dsb-python-function-0.1.0
            helm.sh/chart-name: dsb-python-function
            helm.sh/chart-version: 0.1.0
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: app
                        operator: In
                        values:
                          - pyfn1
                  topologyKey: kubernetes.io/hostname
          containers:
            - env:
                - name: LANG
                  value: en_GB.UTF-8
              envFrom:
                - secretRef:
                    name: secret1
                - configMapRef:
                    name: configMap1
                - configMapRef:
                    name: pyfn1
              image: dsb/py-fn:1.0.0
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8000
                initialDelaySeconds: 10
              name: pyfn1
              ports:
                - containerPort: 8000
                  name: web
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8000
                initialDelaySeconds: 5
              resources:
                limits:
                  cpu: 100m
                  memory: 128Mi
                requests:
                  cpu: 25m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                privileged: false
                readOnlyRootFilesystem: true
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /work
                  name: my-empty-dir
          nodeSelector:
            NodePool: workers
          serviceAccountName: pyfn1-sa
          volumes:
            - emptyDir: {}
              name: my-empty-dir
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/component: Function
        app.kubernetes.io/created-by: https://github.com/dsb-norge/helm-charts
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: Rendering test
        app.kubernetes.io/part-of: Verification stuff
        helm.sh/chart: dsb-python-function-0.1.0
        helm.sh/chart-name: dsb-python-function
        helm.sh/chart-version: 0.1.0
      name: pyfn1
      namespace: NAMESPACE
    spec:
      rules:
        - host: example.com
          http:
            paths:
              - backend:
                  service:
                    name: pyfn1
                    port:
                      number: 8000
                path: /example
                pathType: Prefix
  4: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: pyfn1
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          app: pyfn1
  5: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: pyfn1-service
        app.kubernetes.io/component: Function
        app.kubernetes.io/created-by: https://github.com/dsb-norge/helm-charts
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: Rendering test
        app.kubernetes.io/part-of: Verification stuff
        app.kubernetes.io/version: 1.0.0
        chart-name: dsb-python-function
        chart-version: 0.1.0
        helm.sh/chart: dsb-python-function-0.1.0
        helm.sh/chart-name: dsb-python-function
        helm.sh/chart-version: 0.1.0
      name: pyfn1
      namespace: NAMESPACE
    spec:
      ports:
        - name: web
          port: 8000
      selector:
        app: pyfn1
Minimal manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: python-fn
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: python-fn
      template:
        metadata:
          annotations:
            apparmor.security.beta.kubernetes.io/pod: runtime/default
            checksum: chart-version=0.1.0_config-hash=8a5c5199a12857d35d25c8338c86efcb3024d19849417d84dba5364dd748cdef
          labels:
            app: python-fn
            app.kubernetes.io/component: Function
            app.kubernetes.io/created-by: Helm
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: RELEASE-NAME
            app.kubernetes.io/part-of: RELEASE-NAME
            app.kubernetes.io/version: latest
            azure.workload.identity/use: "false"
            helm.sh/chart: dsb-python-function-0.1.0
            helm.sh/chart-name: dsb-python-function
            helm.sh/chart-version: 0.1.0
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: app
                        operator: In
                        values:
                          - python-fn
                  topologyKey: kubernetes.io/hostname
          containers:
            - env:
                - name: LANG
                  value: en_US.UTF-8
              envFrom: null
              image: python:latest
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8080
                initialDelaySeconds: 10
              name: python-fn
              ports:
                - containerPort: 8080
                  name: web
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8080
                initialDelaySeconds: 5
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 50m
                  memory: 128Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                privileged: false
                readOnlyRootFilesystem: true
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
          nodeSelector:
            NodePool: workers
          serviceAccountName: python-fn-sa
  2: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: python-fn
    spec:
      minAvailable: 0
      selector:
        matchLabels:
          app: python-fn
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: python-fn-service
        app.kubernetes.io/component: Function
        app.kubernetes.io/created-by: Helm
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/part-of: RELEASE-NAME
        app.kubernetes.io/version: latest
        chart-name: dsb-python-function
        chart-version: 0.1.0
        helm.sh/chart: dsb-python-function-0.1.0
        helm.sh/chart-name: dsb-python-function
        helm.sh/chart-version: 0.1.0
      name: python-fn
      namespace: NAMESPACE
    spec:
      ports:
        - name: web
          port: 8080
      selector:
        app: python-fn
Support multiple ingress hosts:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/component: Function
        app.kubernetes.io/created-by: Helm
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/part-of: RELEASE-NAME
        helm.sh/chart: dsb-python-function-0.1.0
        helm.sh/chart-name: dsb-python-function
        helm.sh/chart-version: 0.1.0
      name: python-fn
      namespace: NAMESPACE
    spec:
      rules:
        - host: example.com
          http:
            paths:
              - backend:
                  service:
                    name: python-fn
                    port:
                      number: 8080
                path: /
                pathType: Prefix
        - host: admin.example.com
          http:
            paths:
              - backend:
                  service:
                    name: python-fn
                    port:
                      number: 8080
                path: /secret
                pathType: Prefix
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: python-fn-service
        app.kubernetes.io/component: Function
        app.kubernetes.io/created-by: Helm
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/part-of: RELEASE-NAME
        app.kubernetes.io/version: latest
        chart-name: dsb-python-function
        chart-version: 0.1.0
        helm.sh/chart: dsb-python-function-0.1.0
        helm.sh/chart-name: dsb-python-function
        helm.sh/chart-version: 0.1.0
      name: python-fn
      namespace: NAMESPACE
    spec:
      ports:
        - name: web
          port: 8080
      selector:
        app: python-fn
should render default with PodDisruptionBudget and should match snapshot:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: python-fn
    spec:
      minAvailable: 0
      selector:
        matchLabels:
          app: python-fn
